src/core/server/elasticsearch/deprecations/is_scripting_disabled.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ElasticsearchClient } from '../../elasticsearch';
+
+const scriptAllowedTypesKey = 'script.allowed_types';
+
+export const isInlineScriptingDisabled = async ({
+  client,
+}: {
+  client: ElasticsearchClient;
+}): Promise<boolean> => {
+  const { body: settings } = await client.cluster.getSettings({
+    include_defaults: true,
+    flat_settings: true,
+  });
+
+  // priority: transient -> persistent -> default
+  const scriptAllowedTypes: string[] =
+    settings.transient[scriptAllowedTypesKey] ??
+    settings.persistent[scriptAllowedTypesKey] ??
+    settings.defaults![scriptAllowedTypesKey] ??
+    [];
```

Comment:
I doubt `transient` and `persistent` settings are that much used for things like `script.allowed_types`, but just in case, we follow ES's priority for settings